packages/fraction-model/src/main.jsx

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { withStyles } from '@material-ui/core/styles';
+import AnswerFraction from './answer-fraction';
+import { CorrectAnswerToggle } from '@pie-lib/pie-toolbox/correct-answer-toggle';
+import FractionModelChart from './fraction-model-chart';
+import { AlertDialog } from '@pie-lib/pie-toolbox/config-ui';
+import cloneDeep from 'lodash/cloneDeep';
+
+export class Main extends React.Component {
+  static propTypes = {
+    model: PropTypes.object,
+    session: PropTypes.object,
+    onSessionChange: PropTypes.func.isRequired,
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      session: {
+        ...props.session,
+        answers: (props.session && props.session.answers) || this.generateAnswers(props.model),
+      },
+      showCorrect: false,
+      answerChangeDialog: {
+        open: false,
+        text: '',
+      },
+      fractionModelKey: 0,
+    };
+    this.callOnSessionChange();
+  }
+
+  /**
+   * Function to trigger when session value change for fraction model
+   */
+  callOnSessionChange = () => {
+    const { onSessionChange } = this.props;
+    if (onSessionChange) {
+      onSessionChange(this.state.session);
+    }
+  };
+
+  /**
+   * Function to generate answers for preview selection
+   * @param {object} model contains model object for fraction model
+   * @returns answer model
+   */
+  generateAnswers = (model) => {
+    const answers = {
+      response: [],
+    };
+    if (model.allowedStudentConfig) {
+      (answers.noOfModel = 0), (answers.partsPerModel = 0);
+    }
+    return answers;
+  };
+
+  /**
+   * Function to trigger when answer change from preview
+   * @param {object} newAnswers contains updated answer model
+   */
+  onAnswerChange = (newAnswers) => {
+    let oldSession = cloneDeep(this.state.session);
+    let newSession = {
+      ...this.state.session,
+      answers: newAnswers,
+    };
+    if (newAnswers.response.length > 0) {
+      this.setState({
+        answerChangeDialog: {
+          open: true,
+          oldSession: oldSession,
+          newSession: newSession,
+          text: 'Changing either the Number of Models or Parts per Model will remove added answer. Are you sure you want to continue?',
+        },
+      });
+    } else {
+      this.setState(
+        (state) => ({
+          session: newSession,
+        }),
+        this.callOnSessionChange,
+      );
+    }
+  };
+
+  /*
+   * Function to toggle correct answer
+   * @param {boolean} show contains boolean value to show correct answer
+   * */
+  toggleShowCorrect = (show) => {
+    this.setState({ showCorrect: show });
+    this.setState({ fractionModelKey: this.state.fractionModelKey + 1 });
```

Comment:
should be change to `this.setState({ showCorrect: show, fractionModelKey: this.state.fractionModelKey + 1 });`

Reply:
Updated